### ISSUE TITLE
Bug: Incorrect "TSImportType#options" end when the option key-value contains trailing comma

### ISSUE THREAD

**Issue by JLHwung** [typescript-eslint/typescript-eslint#11474]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&showAST=es&fileType=.tsx&code=DYUwLgBAHgXBYE8AOID2AzCBLAtk1ATmABQBE6qqAdAFYDOqAdqQDQQDeEA7lmABZxOiFHFL0mpCAF82UgJQBuAFBA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
let x: typeof import("foo.json", { with: { type: "json" }, });
```

### ESLint Config

```javascript
{
  "rules": {}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The "TSImportType#options", an `ObjectExpression`, should end at column 60, the last `}` in the source.

### Actual Result

The ObjectExpression currently ends at column 58, the trailing `,` in the source.

### Additional Info

The trailing comma within option has been supported since TS 5.9.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `8.39.1` |
| `TypeScript`                           | `5.9.2` |

**Comment by kirkwaiblinger** [comment]
Hi @JLHwung!

I'm working on this issue. Just for reference, is there a babel issue for supporting the trailing comma? It doesn't seem to be permitted for now... (seemingly the restrictions in https://github.com/babel/babel/pull/17193 are still current?)

**Comment by JLHwung** [comment]
The Babel PR is https://github.com/babel/babel/pull/17465, published in `@babel/parser@7.28.3`.

**Comment by kirkwaiblinger** [comment]
@JLHwung I'm trying in the REPL but not seeing it... am I setting this up wrong? ([link](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&corejs=3.42&code_lz=C4TwDgpgBAKgTgQwJYBskDsDmBhA9gW3wSgF4ol8xc5gAKAcgDNdd6AaKAbygHclgAFgC4uUUJBEAiAFYBnXOkkcAvioCUAbiA&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Ctypescript&version=7.28.3))

**Comment by JLHwung** [comment]
> [@JLHwung](https://github.com/JLHwung) I'm trying in the REPL but not seeing it... am I setting this up wrong? ([link](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&corejs=3.42&code_lz=C4TwDgpgBAKgTgQwJYBskDsDmBhA9gW3wSgF4ol8xc5gAKAcgDNdd6AaKAbygHclgAFgC4uUUJBEAiAFYBnXOkkcAvioCUAbiA&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Ctypescript&version=7.28.3))

Thanks. The setting is correct. The output is empty which is expected because Babel successfully parses and thus stripes it. As for the error you may observe in the editor, the error is not thrown by Babel, but instead from an older typescript version used by the `monaco` editor, we will probably update that editor soon.
